src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2577 @@
+        public void ParenthesizedExpression_GoTo2()
+        {
+            var code =
+@"
+static void Main()
+{
+    int n = 1;
+    switch (n)
+    {
+        case 1:
+            goto case (2+1$$)
+        case 3:
+            break
+        default:
+            break;
+    }
+}
+";
+
+            var expected =
+@"
+static void Main()
+{
+    int n = 1;
+    switch (n)
+    {
+        case 1:
+            goto case (2+1);$$
+        case 3:
+            break
+        default:
+            break;
+    }
+}
+";
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ParenthesizedExpression_Switch()
+        {
+            var code =
+@"
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        switch (i$$)
+        {
+            case 1:
+            case 2:
+            case 3:
+                break;
+        }
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ParenthesizedExpression_Switch2()
+        {
+            var code =
+@"
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        switch (4*(i+2$$))
+        {
+            case 1:
+            case 2:
+            case 3:
+                break;
+        }
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ParenthesizedExpression_Switch3()
+        {
+            var code =
+@"
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        switch (i)
+        {
+            case 1:
+                Console.WriteLine(4*(i+2$$))
+            case 2:
+            case 3:
+                break;
+        }
+    }
+}
+";
+            var expected =
+@"
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        switch (i)
+        {
+            case 1:
+                Console.WriteLine(4*(i+2));$$
+            case 2:
+            case 3:
+                break;
+        }
+    }
+}
+";
+
+            VerifyTypingSemicolon(code,expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ParenthesizedExpression_While()
+        {
+            var code =
+@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        while (i<4$$)
+        {
+            Console.WriteLine(i);
+        }
+    }
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ParenthesizedExpression_While2()
+        {
+            var code =
+@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        int i = 3;
+        while (i<Math.Max(4,5$$))
```

Comment:
>i<Math.Max(4,5$$) [](start = 15, length = 17)

Do we handle this within a do-while statement?

Reply:
We handle it in do-while statements that have other delimiters in the condition.  But the simplest case of do-while, ie `while (n$$ < 5) ` is not handled at this time.